raven/contrib/django/models.py

```diff
@@ -151,66 +155,95 @@ def sentry_exception_handler(request=None, **kwargs):
             warnings.warn('Unable to process log entry: %s' % (exc,))
 
 
-def register_handlers():
-    from django.core.signals import got_request_exception, request_started
-
-    def before_request(*args, **kwargs):
-        client.context.activate()
-    request_started.connect(before_request, weak=False)
+class SentryDjangoHandler(object):
+    def __init__(self, client=client):
+        self.client = client
 
-    # HACK: support Sentry's internal communication
-    if 'sentry' in settings.INSTALLED_APPS:
-        from django.db import transaction
-        # Django 1.6
-        if hasattr(transaction, 'atomic'):
-            commit_on_success = transaction.atomic
+        try:
+            import celery
+        except ImportError:
+            self.has_celery = False
         else:
-            commit_on_success = transaction.commit_on_success
+            self.has_celery = celery.VERSION >= (2, 5)
 
-        @commit_on_success
-        def wrap_sentry(request, **kwargs):
-            if transaction.is_dirty():
-                transaction.rollback()
-            return sentry_exception_handler(request, **kwargs)
+        self.celery_handler = None
 
-        exception_handler = wrap_sentry
-    else:
-        exception_handler = sentry_exception_handler
+    def install_celery(self):
+        from raven.contrib.celery import (
+            SentryCeleryHandler, register_logger_signal
+        )
 
-    # Connect to Django's internal signal handler
-    got_request_exception.connect(exception_handler, weak=False)
+        self.celery_handler = SentryCeleryHandler(client).install()
 
-    # If Celery is installed, register a signal handler
-    if 'djcelery' in settings.INSTALLED_APPS:
-        try:
-            # Celery < 2.5? is not supported
-            from raven.contrib.celery import (
-                register_signal, register_logger_signal)
-        except ImportError:
-            logger.exception('Failed to install Celery error handler')
-        else:
+        # try:
+        #     ga = lambda x, d=None: getattr(settings, 'SENTRY_%s' % x, d)
```

Comment:
need to figure this out, but id prefer it be part of SentryCeleryHandler or not exist by default